src/Planner/PlannerActionsVisitor.cpp

```diff
@@ -243,8 +243,26 @@ class ActionNodeNameHelper
             }
             case QueryTreeNodeType::LAMBDA:
             {
-                auto lambda_hash = node->getTreeHash();
```

Comment:
Why can it differ on initiator and shards?

Reply:
In this scenario, because we have a specified default database in cluster configuration.
So, storage id has empty database on initiator, but the specified one on replica.

Anyway, I think that query tree hash can be reliable only within one replica/shard. Even a small change in hash calculation would break some queries in between versions.

At the same time, I think that lambda execution name is quite reliable in the current implementation. It's basically an execution name for the internal expression (which we assume to be reliable) + (names and types) for lambda parameters.